src/main/java/org/gridsuite/securityanalysis/server/repositories/SubjectLimitViolationRepository.java

```diff
@@ -6,15 +6,62 @@
  */
 package org.gridsuite.securityanalysis.server.repositories;
 
+import jakarta.persistence.criteria.*;
+import org.gridsuite.securityanalysis.server.dto.ResourceFilterDTO;
 import org.gridsuite.securityanalysis.server.entities.SubjectLimitViolationEntity;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 import java.util.List;
 import java.util.UUID;
 /**
  * @author Kevin Le Saulnier <kevin.lesaulnier at rte-france.com>
  */
 
-public interface SubjectLimitViolationRepository extends JpaRepository<SubjectLimitViolationEntity, UUID> {
-    List<SubjectLimitViolationEntity> findByResultIdOrderBySubjectId(UUID resultUuid);
+public interface SubjectLimitViolationRepository extends CommonLimitViolationRepository<SubjectLimitViolationEntity>, JpaRepository<SubjectLimitViolationEntity, UUID>, JpaSpecificationExecutor<SubjectLimitViolationEntity> {
```

Comment:
for the consistency with other repos interfaces, could you please add @Repository, even so technically not necessary :)

Reply:
fixed